app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsActivityTest.kt

```diff
@@ -133,11 +149,40 @@ class MyDownloadsActivityTest {
     assertThat(screenName).isEqualTo(ScreenName.MY_DOWNLOADS_ACTIVITY)
   }
 
+  @Test
+  @RunOn(TestPlatform.ESPRESSO)
+  fun testMyDownloadsActivity_pressBack_checkOpensHomeActivity() {
+    ActivityScenario.launch(MyDownloadsActivity::class.java).use {
+      pressBack()
+      intended(hasComponent(HomeActivity::class.java.name))
+      it.onActivity { activity ->
+        assertThat(
+          activity.intent.extractCurrentUserProfileId().internalId
+        ).isEqualTo(-1)
+      }
+    }
```

Comment:
Here and below,

```suggestion
    pressBack()
    intended(IntentMatchers.hasComponent(HomeActivity::class.java.name))
    hasExtraWithKey(PROFILE_ID_INTENT_DECORATOR)
    // maybe also intended(hasProtoExtra(PROFILE_ID_INTENT_DECORATOR, profileId))
```
`onActivity { }` should only be used when verifying things that happen after an activity is launched. In this case, pressing back should kill the current activity, causing failure if you try to run `hasExtraWithKey(PROFILE_ID_INTENT_DECORATOR)`.

In your case I can tell that you are trying to test if  `MyDownloadsActivity.intent`(the launched scenario to which onActivity belongs), has that profileId. However, that is not important to verify here. 

In terms of unit tests, it is technically supposed to be tested in an activity that launches MyDownloadsActivity, such as the `NavigationDrawerActivity`, where we would need to verify that the intent has all the expected params.

As per my suggestions, here we are interested in knowing if the home activity intent is launched, with the expected parameters.

This applies to all intent-related tests in this PR.

Reply:
Thanks for the detailed explanation, updated per suggestion.